runtime-sdk/src/modules/consensus_accounts/mod.rs

```diff
@@ -599,6 +602,36 @@ impl<Consensus: modules::consensus::API> Module<Consensus> {
         state::get_undelegations(args.to)
     }
 
+    #[handler(call = "consensus.SharesToTokens", internal)]
+    fn internal_shares_to_tokens<C: Context>(
+        ctx: &C,
+        args: types::SharesToTokens,
+    ) -> Result<u128, Error> {
+        let params = Self::params();
+        <C::Runtime as Runtime>::Core::use_tx_gas(params.gas_costs.shares_to_tokens)?;
+
+        let account = Consensus::account(ctx, args.address)?;
+        let pool = match args.pool {
+            types::SharePool::Active => &account.escrow.active,
+            types::SharePool::Debonding => &account.escrow.debonding,
+            _ => return Err(Error::InvalidArgument),
+        };
+
+        let tokens = args
+            .shares
+            .checked_mul(
+                u128::try_from(pool.balance.clone()).expect("pool balance should fit into u128"),
+            )
+            .expect("shares * total_amount should not overflow")
+            .checked_div(
+                u128::try_from(pool.total_shares.clone())
+                    .expect("pool total_shares should fit into u128"),
+            )
+            .expect("total_shares should not be zero");
```

Comment:
This should not crash the runtime. What if one provides an account that does not have anything delegated? It should either return zero or an error (an error is probably better).